avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroSerDeConfig.java

```diff
@@ -79,7 +88,9 @@
 
   public static ConfigDef baseConfigDef() {
     return new ConfigDef()
-        .define(SCHEMA_REGISTRY_URL_CONFIG, Type.LIST,
+        .defineAlternative("registry", MOCK_SCHEMA_REGISTRY_SCOPE_CONFIG, Type.STRING, null,
+                Importance.LOW, MOCK_SCHEMA_REGISTRY_SCOPE_DOC)
+        .defineAlternative("registry", SCHEMA_REGISTRY_URL_CONFIG, Type.LIST, null,
```

Comment:
do we actually need this alternative stuff? Couldn't you just use a special protocol in the SR url to indicate you want a mocked, local SR? e.g. `local://` or something like that?

Reply:
Yeah, we could. I was thinking `mock://`, but at the same time, I worried that it might be too "magic". I.e., insufficient differentiation between testing and production configuration.

But I haven't had a chance to discuss it with anyone. WDYT?

Reply:
Oh, also, this "scope" stuff is new. I started off with passing in a class for the mock registry, so it wasn't possible to re-use the same config, which led me to implement the "alternative" support.

Now that they're both semantically similar strings, maybe the "mock url" think makes more sense on balance.

Reply:
yeah `mock://scope` could work

mentioned in other pr, but main concern for me is that I don't know how things generalize. only have this one example currently, so hard to say which way is better

Reply:
Ok, the fact that this is apparently the first such use case in many years gave me a little bit of pause anyway.

It seems like the mock url approach would work and is semantically sound, and is a less invasive change. I'll go ahead and give it a shot, and maybe we can just close the config PR.